Fix typos when setting the attributes of JarHash

aptmac opened the pull request:
I ran into this while testing out retrieval of jar hash information (using the test resource files as input). There are some typos in the attribute names when setting the jar hash information from a jvm instance.

The jar information can look something like:
```
"jars" : [ {
      "name" : "",
      "version" : "",
      "attributes" : {
        "sha1Checksum" : "",
        "Implementation-Vendor" : "",
        "groupId" : "",
        "sha256Checksum" : "",
        "artifactId" : "",
        "sha512Checksum" : "",
        "Implementation-Vendor-Id" : "",
        "version" : ""
      }
} ]
```

And here's an example of what information comes back before & after this commit:

Before:
```
{
  id=ba834aa9-8ebb-4a1b-8ea1-520268a060f2,
  instance=JvmInstance{[..]},
  name='jboss-modules.jar',
  groupId='',
  vendor='',
  version='1.12.0.Final-redhat-00001', 
  sha1Checksum='',
  sha256Checksum='',
  sha512Checksum=''
}
```

After:
```
{
  id=5171a2af-b782-49f6-a59b-88f7f957d6fc,
  instance=JvmInstance{[..]},
  name='jboss-modules.jar',
  groupId='org.jboss.modules',
  vendor='JBoss by Red Hat',
  version='1.12.0.Final-redhat-00001', 
  sha1Checksum='82a8c99551533f4448675f273665cb6d7b750511',
  sha256Checksum='d808a03cf5f844f0d1cf52b340a5c4ad836c052e1288a9b5ac8ca6df6ae9e000',
  sha512Checksum='4c85d6f74cb8a34dca6748873f9b38457c04f253c4b7d1e088010d13eec7021145b338979adf8991f54c2b6241da7d350c2cfac849e603c286aa5fb13edf560f'
}
```

kittylyst reviewed (changes requested):
Can you add a test case for this, please? I *think* I know what it's doing, but I'm not 100% sure as it is.

aptmac commented:
Sure I can try to figure out a test for this. I'll try to explain it a bit more in depth too.

It's a mismatch when reading the jar json data and trying to extract the data. `Utils.jarHashOf()` is given the raw json, for example let's use [`test17.json`](https://github.com/RedHatInsights/insights-runtimes-inventory/blob/main/events/src/test/resources/test17.json#L65) [[0]](https://github.com/RedHatInsights/insights-runtimes-inventory/blob/main/events/src/test/resources/test17.json#L65) for example, which looks something like:
```
{
   "name" : "jackson-core-2.14.2.jar",
   "version" : "2.14.2",
   "attributes" : {
     "sha1Checksum" : "f804090e6399ce0cf78242db086017512dd71fcc",
     "Implementation-Vendor" : "FasterXML",
     "groupId" : "com.fasterxml.jackson.core",
     "sha256Checksum" : "b5d37a77c88277b97e3593c8740925216c06df8e4172bbde058528df04ad3e7a",
     "artifactId" : "jackson-core",
     "sha512Checksum" : "208953cf72691b3bab1ca2f9a045cccd3d75c46c90099ba884fbf16bc3e45045e5cd014d2c68c96e8a8826f87a5f0e18cb34b35e7ef03348953869c21e197367",
     "Implementation-Vendor-Id" : "com.fasterxml.jackson.core",
     "version" : "2.14.2"
}
```

`Utils.jarHashOf()` gets this json and at the moment and tries a string match to pick out: `group_id`, `vendor`, `sha1_checksum`, `sha256_checksum`, `sha512_checksum`. 

While this how the attributes names look in the database [[1]](https://github.com/RedHatInsights/insights-runtimes-inventory/blob/main/events/src/main/resources/db/migration/V1.0.0__jvm_instance.sql#L24), in the incoming json they're written as `groupId`, `Implementation-Vendor`, `sha1Checksum`, `sha256Checksum`, and `sha512Checksum` as seen in the above json. As a result, `Utils.jarHashOf()` doesn't fetch these attributes, and they're written to the db with an empty string.

[0] https://github.com/RedHatInsights/insights-runtimes-inventory/blob/main/events/src/test/resources/test17.json#L65
[1] https://github.com/RedHatInsights/insights-runtimes-inventory/blob/main/events/src/main/resources/db/migration/V1.0.0__jvm_instance.sql#L24

aptmac commented:
Rebased on top of upstream/main

Added a quick unit test to verify that the jar hash object has it's attributes written and are as expected.

I brought this same unit test on upstream/main and it fails with:
```
[ERROR] Failures: 
[ERROR]   EventConsumerTest.testSettingJarHashAttributes:114 expected: <JBoss by Red Hat> but was: <>
[ERROR]   EventConsumerTest.testSettingJarHashAttributes:115 expected: <82a8c99551533f4448675f273665cb6d7b750511> but was: <>
[ERROR]   EventConsumerTest.testSettingJarHashAttributes:116 expected: <d808a03cf5f844f0d1cf52b340a5c4ad836c052e1288a9b5ac8ca6df6ae9e000> but was: <>
[ERROR]   EventConsumerTest.testSettingJarHashAttributes:119 expected: <4c85d6f74cb8a34dca6748873f9b38457c04f253c4b7d1e088010d13eec7021145b338979adf8991f54c2b6241da7d350c2cfac849e603c286aa5fb13edf560f> but was: <>
```
